src/modules/launcher/Wox.Test/Plugins/Win32Tests.cs

```diff
@@ -333,5 +333,48 @@ public void RunCommands_ShouldNotBeFiltered_OnExactMatch(string query)
             // the query matches the name (cmd) and is therefore not filtered (case-insensitive)
             Assert.IsTrue(cmd_run_command.QueryEqualsNameForRunCommands(query));
         }
+
+        // Mock Win32 class to test the Result properties
+        public class mockWin32 : Win32
+        {
+            protected override int Score(string query)
+            {
+                // To prevent the result from being filtered, the score must be a positive integer
+                const int scoreGreateThanZero = 1;
+                return scoreGreateThanZero;
+            }
+
+            // Stubbing out the StringMatcher.Fuzzysearch() functionality
+            protected override List<int> GetMatchingData(string query)
+            {
+                return null;
+            }
+        }
+
+        [TestCase("Name", "NamePrefixInDescription")]
+        [TestCase("ignoreName", "ignoreDescription")]
+        public void Win32Apps_ShouldSetNameAsTitle_WhileCreatingResult(string Name, string Description)
```

Comment:
I'm not sure I understand your question because we cannot directly use a Win32 program to test the functionality as we need to mock the functions such as `Score` and `GetMatchingData`. Failing to do so, would throw errors. I have tested Win32 applications by creating a derived class of Win32. So to answer your question, Win32 applications are already being tested.

Reply:
It looks like the fuzzy match requires an instance of itself which is set during powertoys run startup. The following test will work : 
```
// Arrange
 var mock = new Mock<IPublicAPI>();
 mock.Setup(x => x.GetTranslation(It.IsAny<string>())).Returns(It.IsAny<string>());
 StringMatcher.Instance = new StringMatcher();

 // Act
var result = chrome.Result("chrome", mock.Object);

// Assert
Assert.IsTrue(result.Title.Equals(chrome.Name));
```
Note : You will need to make internal of `Wox.infrastructure` visible to `Wox.test` as done [here](https://github.com/microsoft/PowerToys/pull/4739/files#diff-6d32f5c41192cbb8fc30b7d828bf6cadR61).

Reply:
Nice catch! I'll do that.